examples/twittermap/web/public/javascripts/mapresultcache/services.js

```diff
@@ -0,0 +1,82 @@
+/*
+ * Module to cache map result data. Common services module communicates with
+ * middleware and updates resultcache module.
+ */
+'use strict';
+angular.module('cloudberry.mapresultcache', ['cloudberry.common'])
+  .service('MapResultCache', function () {
+
+    // The key-value store that stores mapResults
+    var store = new HashMap();
+    // To check if keyword in query changed
+    var currentKeywords = [""];
+    // To check if time range in query changed
+    var date = new Date();
+    date.setDate(date.getDate() - 1);
+    var currentTimeRange = {
+      start: new Date(),
+      end: date
+    };
+    // Prefix for geoIds to make key unique
+    var prefix = Object.freeze({
+      state: 'S',
+      county: 'C',
+      city: 'I'
+    });
+    const INVALID_VALUE = 0;
+
+    // Check keyword, time range and the cache store
+    // Return the geoIds not in store
+    this.returnGeoIdsNotInCache = function (keywords, timeInterval, geoIds, geoLevel) {
```

Comment:
I think the `getNotInCacheGeoId` would be a better name.

Reply:
and can you make it as a normal way of writing the function? e.g.
```
 function getNotInCacheGeoId(keywords, timeInterval, geoIds, geoLevel) {
...
}
```

Reply:
If we declare it as a regular function, it is not accessible in the common module.